File: 0164-maximum-gap/0164-maximum-gap.java
class Solution {
    public int maximumGap(int[] nums) {
        Stack<Integer> res=new Stack<>();
        if(nums.length<2) return 0;
        Arrays.sort(nums);
        int maxs=0;
        for(int i=0;i<nums.length-1;i++){
            maxs=Math.max(maxs,nums[i+1]- nums[i]);
        }
        return maxs;
    }
}

File: 0164-maximum-gap/0164-maximum-gap.cpp
class Solution {
public:
    int maximumGap(vector<int>& nums) {
        int n=nums.size();
        sort(nums.begin(),nums.end());
        if(n<2) return 0;
        int maxs=0;
        for(int i=0;i<n-1;i++){
            maxs=max(maxs,nums[i+1]- nums[i]);
        }
        return maxs;
    }
};